### PR TITLE
Add --no-install-recommends to apt-get install

### PR DESCRIPTION
**Pull request by ldorau**
It decreases size of Ubuntu Docker image from 418 MB to 368 MB.

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/pmemkv/267)

### CODE REVIEW

**Comment by ldorau** [comment]
@marcinslusarz please review

File: utils/docker/images/Dockerfile.ubuntu-18.04 [hunk 1]
```diff
@@ -39,16 +39,20 @@ FROM ubuntu:18.04
 MAINTAINER lukasz.stolarczuk@intel.com
 
 # Update the Apt cache and install basic tools
-RUN apt-get update \
- && DEBIAN_FRONTEND=noninteractive apt-get install -y software-properties-common \
+RUN apt-get update --no-install-recommends
+RUN DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+	apt-utils
```
Comment (marcinslusarz) [inline utils/docker/images/Dockerfile.ubuntu-18.04]: I discussed this with @kilobyte. This is not needed.

File: utils/docker/images/Dockerfile.ubuntu-18.04 [hunk 2]
```diff
@@ -66,6 +70,7 @@ RUN apt-get update \
 	pkg-config \
 	rapidjson-dev \
 	ruby \
+	software-properties-common \
```
Comment (marcinslusarz) [inline utils/docker/images/Dockerfile.ubuntu-18.04]: shouldn't be needed

File: utils/docker/images/Dockerfile.ubuntu-18.04 [hunk 3]
```diff
@@ -39,16 +39,20 @@ FROM ubuntu:18.04
...
 # Update the Apt cache and install basic tools
-RUN apt-get update \
- && DEBIAN_FRONTEND=noninteractive apt-get install -y software-properties-common \
+RUN apt-get update --no-install-recommends
+RUN DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+	apt-utils
+RUN DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
 	autoconf \
 	automake \
+	build-essential \
```
Comment (marcinslusarz) [inline utils/docker/images/Dockerfile.ubuntu-18.04]: this pulls in gcc and g++, so there's no need to list them separately

**Comment by marcinslusarz** [comment]
> whois and doxygen are also very likely not needed

sorry, "whois" is needed by mkpasswd used a few lines later

**Comment by lukaszstolarczuk** [comment]
perhaps need to check if Coverity build is working fine with this truncated list of packages

**Pull request merged**